README.md

```diff
@@ -1 +1,679 @@
+```
+
+As you can see, both the empty element at the beginning and the incorrect element at index 3 are None. 
+
+I want to add one thing in code here even though it is technically not needed - tokenize currently returns our tokens as an array. I prefer to work with tuples in f#, because it comes with a few benefits (remember the `||>` from above), so we will convert it. We can easily do this with the line `fun line -> Some(line[0], line[1])`, but we can't just pipe it in this case. If we write>
+
+```fsharp
+let tokenize(line: string) =
+    line
+    |> split '='
+    |> fun line -> Some(line[0], line[1])
+```
+
+the compiler will complain about 
+```txt
+The type 'Option<_>' does not define the field, constructor or member 'Item'.
+```
+
+The compiler is trying to tell us that line is not an array but an Option. Remember the Some/None Maybe that the split function returns. To get rid of the compiler error, we would have to add `.Value` on the line to access the content of the Maybe:
+
+```fsharp
+let tokenize(line: string) =
+    line
+    |> split '='
+    |> fun line -> Some(line.Value[0], line.Value[1])
+```
+
+This, however, will fail during runtime as the line might be None and thus not have a Value. To make sure we are doing this right, we must bind the value first. Binding in that context means checking it and only executing it when it is some.
+
+We can add this function before the `fun` so that the result will look like this:
+
+```fsharp
+let tokenize(line: string) =
+    line
+    |> split '='
+    |> Option.bind (fun line -> Some(line[0], line[1]))
+```
+
+Rerunning the code will lead to the following output:
+
+```sh
+❯ dotnet fsi pipes.fsx
+seq [None; Some ("ELEMENT01", "3"); Some ("ELEMENT04", "4"); None; ...]
+```
+
+As you can see, the elements have changed from `[|" ELEMENT01"; "3" |]` to `("ELEMENT01", "3")`, they are tuples now. 
+
+As we will need the binding more often, we will add an infix operator that does the binding for us in the future. The code becomes
+
+```fsharp
+let (>>=) m f = Option.bind f m
+
+let tokenize(line: string) =
+    line
+    |> split '='
+    >>= fun line -> Some(line[0], line[1])
+```
+
+As we added the new `>>=` operator, we no longer have to write `Option.bind`. Coincidentally, the implementation of `>>=` is almost the same as for `|>` - the only change is that it calls `Option.bind` before calling the passed function.
+
+We haven't talked about the javascript implementation of the pipe until this point. And we won't for some time. But do note that the pipe in f# is merely a starting point for us on our quest to compose functions. It's not just syntactic sugar to avoid reading inside out; it stands on the shoulder of the giant that implicitly and nicely asks us to think in functions and dive into them just as popping up. And naturally, by the laws and the ecosystem that we have around this, we are pushed into a direction that will prevent us later from making mistakes. 
```

Comment:
Jop, the "why we don't have that issue in javascript without the pipe operator", that you asked about later goes in the same direction. I also had this discussions with some friends on our Signal chat. I will add a paragraph on the core message, maybe as TL;DR in the beginning, hoping this will clear things up